homeassistant/components/lifx/coordinator.py

```diff
@@ -186,60 +190,85 @@ def async_get_entity_id(self, platform: Platform, key: str) -> str | None:
+        device = self.device
+        if self.device.host_firmware_version is None:
+            methods.append(device.get_hostfirmware)
+        if self.device.product is None:
+            methods.append(device.get_version)
+        if self.device.group is None:
+            methods.append(device.get_group)
+        assert methods, "Device info already populated"
+        await async_multi_execute_lifx_with_retries(
+            methods, DEFAULT_ATTEMPTS, OVERALL_TIMEOUT
+        )
 
-            if lifx_features(self.device)["hev"]:
-                await self.async_get_hev_cycle()
+    @callback
+    def _async_build_color_zones_update_requests(self) -> list[Callable]:
+        """Build a color zones update request."""
+        device = self.device
+        return [
+            partial(device.get_color_zones, start_index=zone)
+            for zone in range(0, len(device.color_zones), 8)
+        ]
 
-            if lifx_features(self.device)["infrared"]:
-                await async_execute_lifx(self.device.get_infrared)
+    async def _async_update_data(self) -> None:
+        """Fetch all device data from the api."""
+        device = self.device
+        if (
+            device.host_firmware_version is None
+            or device.product is None
+            or device.group is None
+        ):
+            await self._async_populate_device_info()
+
+        num_zones = len(device.color_zones) if device.color_zones is not None else 0
+        features = lifx_features(self.device)
+        is_extended_multizone = features["extended_multizone"]
+        is_legacy_multizone = not is_extended_multizone and features["multizone"]
+        update_rssi = self._update_rssi
+        methods: list[Callable] = [self.device.get_color]
+        if update_rssi:
+            methods.append(self.device.get_wifiinfo)
+        if is_extended_multizone:
+            methods.append(self.device.get_extended_color_zones)
+            methods.append(self.device.get_multizone_effect)
+        elif is_legacy_multizone:
+            methods.append(self.device.get_extended_color_zones)
+            methods.extend(self._async_build_color_zones_update_requests())
+        if features["hev"]:
+            methods.append(self.device.get_hev_cycle)
+        if features["infrared"]:
+            methods.append(self.device.get_infrared)
+
+        responses = await async_multi_execute_lifx_with_retries(
+            methods, MAX_ATTEMPTS_PER_UPDATE_REQUEST_MESSAGE, MAX_UPDATE_TIME
+        )
+        # device.mac_addr is not the mac_address, its the serial number
+        if device.mac_addr == TARGET_ANY:
+            device.mac_addr = responses[0].target_addr
+
+        if update_rssi:
+            # We always send the rssi request second
+            self._rssi = int(floor(10 * log10(responses[1].signal) + 0.5))
+
+        if is_extended_multizone:
+            self.active_effect = FirmwareEffect[self.device.effect.get("effect", "OFF")]
```

Comment:
Why is this only for extended multizone devices? Legacy multizone devices support effects too.

Reply:
refactoring error. fixed